### ISSUE TITLE
setting trust for MPType.Unspecified mods to true doesn't allow joining vanilla rooms as client with such mods

### ISSUE THREAD

**Issue by heaventwig** [Void-Crew-Modding-Team/VoidManager#5]
Setting the trust override in the F5 menu to "true" doesn't enable joining unmodded rooms as a client with MPType.Unspecified mods running.

## frequency
always

## steps to duplicate
- install eg Maverik's BetterScoop mod (which is only effective if run on the host, but doesn't break the game when running on a client where the host doesn't have it).
- install VoidManager
- run Void Crew
- hit F5, tap the button to set options for VoidManager, and set trust override to true
- try to join a vanilla room

**Comment by DragonFire47** [comment]
That particular feature is meant for hosts, to determine whether they should allow players to join with mods that haven't specified their MPType to VoidManager.

When a client joins a game with unspecified mods, VoidManager doesn't know if it's a client-side mod, so it decides not to trust it.

If the client you're joining has VoidManager installed, they can enable the trust feature and allow you to join with the mod, or you could disable the mod via your ModManager before launching the game.

TL;DR
This is intentional behavior.

**Comment by heaventwig** [comment]
I'm thinking maybe we're talking past each other here.

I'm wanting to be able to load Void Crew with my standard pile of mods, which includes a bunch that don't have specified types. I want then to go back and forth between hosting my own (modded) rooms and joining others' (usually vanilla) rooms without having to cycle Void Crew. (I'm the only one in any of these rooms who has VoidManager loaded.)

These other mods I have loaded don't have any trouble with this practice. But when I have VoidManager loaded, I can't have e.g. BetterScoop loaded and join someone else's (vanilla, unmodded) room.

**Comment by heaventwig** [comment]
Maybe this would be clearer:
- you seem to be addressing the case where I'm hosting, and I can set VoidManager to allow or disallow clients joining with MPType.Unspecified mods.
- the issue I'm attempting to report is the case where I'm joining as a client with VoidManager running, and it prevents me joining other people's rooms because I myself have loaded mods that haven't implemented VoidManager's MPType specification system.

**Comment by heaventwig** [comment]
Given that the error messages from VoidManager when I'm prevented from joining in this way include the names of the mods I'm running, I'm thinking VoidManager must be able to get a list of running mods. It has to do this to cast their MPType as Unspecified. Could there be an interface element added to allow the user to — possibly even persistently, via VoidManager's config file — specify a type for each of their mods? Then I could
- respecify Voider Crew as breaky or not, each time I adjust the number of players
- persistently specify Maverik's mods as vanilla-friendly

If there were also a "persistence" checkbox, I could leave Voider Crew unchecked to prevent downstream accidental badness, but check the box beside Maverik's mods to keep those settings around.

**Comment by DragonFire47** [comment]
A way to mark mods in such a manner might be in the realm of acceptability for us. I'm thinking it could be done through a text file or the config file, by adding the names of the mods.

**Comment by heaventwig** [comment]
Yes, adding whatever identifier VoidManager sees for the other unspecified mods to the config file—along with a type specification—is exactly what I had in mind.

**Comment by DragonFire47** [comment]
Giving a small update, currently planning on making mods not configured by the author manually configurable with the next VoidManager update. Not sure when it'll be released, but probably next weekend at the latest.

It'll be managed via VoidManager's BepInEx config file, with the option to configure mods as client-side or all-required. Users will specify the mod to voidmanager via the mod GUID or name (internally set by mod authors).

**Comment by heaventwig** [comment]
That's great, and I imagine it will take care of this issue!

I'm curious, though — any reason you're not including the ability to mark a mod with any of the MPType specifiers? Seems like limiting it to all and client would make it less functional than it could be.

**Comment by DragonFire47** [comment]
Ideally, the mod developers would be configuring their mod for VoidManager. Additionally, the only MPType a user can't apply to mods with this setup is a debug feature, so the other 3 are usable.

**Comment by DragonFire47** [comment]
This feature has been provided with update 1.0.6.  A description with examples can be found in the config file, and the input you'll need to specifically configure the better scoop mod is 'Better Scoop:Client'. You can check it's effectiveness in the F5 menu. Should more MPTypes be added in the future, they will most likely be made configurable.
